### DISCUSSION TITLE
Specifying an event observable

### DISCUSSION THREAD

**Discussion by fkonr** [AMICI-dev/AMICI#1807] [Q&A] [answered]
Hi,
I am currently using the amici toolbox for MATLAB and decided to change to the python-based version. I created an SBML version of my model which contains an event. The simulations using python work fine, i.e. the event is triggered and the specified state changes are applied. However, I would like to define an output for the event. For the MATLAB version, one could use the amievent() function to define the variable "z". How can this be achieved in the python version? It seems that it is not possible to define the event output in the SBML file. I found in the API references the "amici.ode_model.Event" class, but I am not sure how I would use it to add to the model a new event or modify the one that was already defined in the SBML file.
I would highly appreciate if you could help me with that!

Thanks. Best, Fabian

**Comment by FFroehlich** [comment, marked as answer]
Hey Fabian,

thank you for your interest in AMICI. Unfortunately, we haven't implemented support for event observables in python yet. I would generally encourage using the python version and will look into implementing support next week.

Best,
Fabian

**Reply by fkonr** [reply]
Hi Fabian,
thanks for your quick reply. This would be really great if event observables would be implemented in the python version as well. Thanks!

Best,
Fabian

**Reply by FFroehlich** [reply]
Sorry, this took so long, but I am finally able to allocate some time working on this!

**Reply by FFroehlich** [reply]
I have implemented support in https://github.com/AMICI-dev/AMICI/pull/1845, so the feature should hopefully be included in the next release.

**Reply by fkonr** [reply]
Great! Thanks a lot.

**Reply by dweindl** [reply]
Event observables via Python are supported in [AMICI 0.12.0](https://github.com/AMICI-dev/AMICI/releases/tag/v0.12.0). Now also available on pypi.
